sufia-models/app/jobs/active_fedora_pid_based_job.rb

```diff
@@ -1,22 +1,7 @@
-class ActiveFedoraPidBasedJob
-  def queue_name
-    :pid_based
-  end
-
-  attr_accessor :pid
-
-  def initialize(pid)
-    self.pid = pid
-  end
-
-  def object
-    @object ||= ActiveFedora::Base.find(pid)
-  end
-
-  alias_method :generic_file, :object
-  alias_method :generic_file_id, :pid
-
-  def run
-    raise RuntimeError, "Define #run in a subclass"
+class ActiveFedoraPidBasedJob < ActiveFedoraIdBasedJob
+  extend Deprecation
+  def self.extended(document)
+    Deprecation.warn ActiveFedoraPidBasedJob, "ActiveFedoraPidBasedJob is deprecated; use ActiveFedoraIdBasedJob instead."
```

Comment:
If you make this change to the 5.x branch, you could go ahead and remove this in 6.0.